src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -453,39 +652,60 @@ class CVulkanLogicalDevice final : public ILogicalDevice
 
     bool bindImageMemory(uint32_t bindInfoCount, const SBindImageMemoryInfo* pBindInfos) override
     {
+        bool anyFailed = false;
         for (uint32_t i = 0u; i < bindInfoCount; ++i)
         {
-            if (pBindInfos[i].image->getAPIType() != EAT_VULKAN)
+            const auto& bindInfo = pBindInfos[i];
+
+            if ((bindInfo.image->getAPIType() != EAT_VULKAN) || (bindInfo.memory->getAPIType() != EAT_VULKAN))
                 continue;
-            VkImage vk_image = static_cast<const CVulkanImage*>(pBindInfos[i].image)->getInternalObject();
 
-            // if (pBindInfos[i].memory->getAPIType() != EAT_VULKAN)
-            //     continue;
-            VkDeviceMemory vk_deviceMemory = static_cast<const CVulkanMemoryAllocation*>(pBindInfos[i].memory)->getInternalObject();
+            CVulkanImage* vulkanImage = static_cast<CVulkanImage*>(bindInfo.image);
+            vulkanImage->setMemoryAndOffset(
+                core::smart_refctd_ptr<IDriverMemoryAllocation>(bindInfo.memory),
+                bindInfo.offset);
```

Comment:
remember that its also an error to bind memory more than once to an image (unless the image is sparse, but then its a sparse bind request which is something completely different)